pages.ts

```diff
@@ -83,3 +94,140 @@ export async function loadData(
 
   return { ...pageData, components: componentsWithData };
 }
+
+export const fetchPageFromPathname = async (
+  path: string,
+  siteId: number
+): Promise<PageWithParams | null> => {
+  const { data: pages, error } = await getSupabaseClient()
```

Comment:
I liked the overall idea. However, fetching all routes is really a performance bottleneck. 

A possible trick in here is to go to the database twice, one looking for an exact match and another one filtering dynamic paths. For this to work, dinamic pages must be tagged like so. We could do this during the save process.

Reply:
Ok, I tried implementing this and got a bit upset because Supabase's can't reuse pieces of query, and we'd have to repeat some stuff. This made me think more about this subject...

The number of production pages should be less than 50, and this is already filtered by the `flag: null, archived: false`. In that case, I suspect that the difference is minimal here, and we'd add an overhead for the dynamic pages (which is ~70% of the store).

Anyway: I'll create a separate task to investigate this problem, possibly implementing a more drastic caching strategy for production. I don't think it's worth going to the database for every page.

Reply:
> The number of production pages should be less than 50, and this is already filtered by the flag: null, archived: false. In that case, I suspect that the difference is minimal here, and we'd add an overhead for the dynamic pages (which is ~70% of the store).

I'm not sure about this one. We have to take into account redirects that can be possibly thousands of them. 

anyway, yeah, we can totally deal with this in the future!

Reply:
https://deco.fibery.io/Product/All-tasks-114#Tasks/20a457e0-5b86-11ed-93ab-f5261eb684bd